src/haven/MapView.java

```diff
@@ -1722,6 +1722,50 @@ protected void nohit(Coord pc) {
         return (new Object[0]);
     }
 
+    private interface GobCheck {
+        boolean check(Gob gob);
+    }
+    
+    private class GobCheckAtk implements GobCheck {
+        public boolean check(Gob gob) {
+            if (gob == null || gob.type == null) {
+                return false;
+            }
+            Gob.Type type = gob.type;
+            switch (type) {
+            case PLAYER:
+                return Config.proximityaggro && !gob.isFriend();
+            case MAMMOTH:
+            case BAT:
+            case EAGLE:
+            case MOB:
+            case BEAR:
+            case LYNX:
+            case WILDGOAT:
+            case TROLL:
+            case WALRUS:
+            case TROUGH:
+                return Config.proximityaggroanimal;
+            default:
+                return false;
+            }
+        }
+    }
+
+    private class GobCheckLift implements GobCheck {
+        public boolean check(Gob gob) {
+            //TODO
```

Comment:
This will be a huge headache to implement properly. Will need to keep object lists and maintain it... same for GobCheckChase